diff --git src/gatekeeperservice.ads src/gatekeeperservice.ads
index 835c185..22be8cf 100644
--- src/gatekeeperservice.ads
+++ src/gatekeeperservice.ads
@@ -7,6 +7,7 @@ package GateKeeperService is
   task GateKeeper is
       -- accept Food_Packs from interplanetary vesssels.
       entry acceptMessage( newFood: in Food_Pack );
+      entry acceptMessageFront (newFood : in Food_Pack);

       --Allow sals to retrive the repackaged Food_Packs.
       entry retrieveMessage( newFood: out Food_Pack; availableForShipment: out Boolean );

